compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/SchemaGenerator.java

```diff
@@ -350,7 +354,24 @@ private void getTypesFromInterface(String typeName, Type interfaceType) {
             }
 
             interfaceType.addPossibleType(implementedType);
-            implementedType.addInterface(interfaceType);
+            addTransitiveImplementationsToInterface(interfaceType, implementedType);
+            addSuperInterfacesToImplementation(implementedType);
+        }
+        visitedInterfaces.remove(interfaceType);
+    }
+
+    private void addTransitiveImplementationsToInterface(Type interfaceType, Type implementedType) {
```

Comment:
How do we handle cyclic references or self-implementing interfaces [issues](https://spec.graphql.org/October2021/#sel-FAHbhBLCAACElB5xc)?

Reply:
This is automatically handled by the compiler. An object can't cyclicly include it self
ex: compiler throw error for the below code
```
public type Node distinct service object {
    *Named;
    resource function get id() returns int;
    resource function get name() returns string;
};

public type Named distinct service object {
    *Node;
    resource function get id() returns int;
    resource function get name() returns string;
};
```